extra_data/tests/test_keydata.py

```diff
@@ -284,6 +285,22 @@ def test_single_value(mock_sa3_control_data, monkeypatch):
     np.testing.assert_equal(intensity.as_single_value(rtol=1), np.median(data))
 
 
+def test_ndarray_out(mock_sa3_control_data):
+    f = H5File(mock_sa3_control_data)
+    beamview = f['SA3_XTD10_IMGFEL/CAM/BEAMVIEW2:daqOutput', 'data.image.dims']
+
+    # Get data and get a new array.
+    buf_new = beamview.ndarray()
+
+    # Get data into a pre-allocated array.
+    buf_in = np.empty(  # Do not use .zeros here to get random data!
```

Comment:
If you want a test that doesn't need this caveat, the Basler cameras in the mock data actually have their `dims` array filled in.

Reply:
Thanks, switched the source.